cluster/service.go

```diff
@@ -154,7 +160,32 @@ func (s *Service) processWriteShardRequest(buf []byte) error {
 		return err
 	}
 
-	if err := s.TSDBStore.WriteToShard(req.ShardID(), req.Points()); err != nil {
+	err := s.TSDBStore.WriteToShard(req.ShardID(), req.Points())
+
+	// We may have received a write for a shard that we don't have locally because the
+	// sending node may have just created the shard (via the metastore) and the write
+	// arrived before the local store could create the shard.  In this case, we need
+	// to check the metastore to determine what database and retention policy this
+	// shard should reside within.
+	if err == tsdb.ErrShardNotFound {
+
+		// Query the metastore for the owner of this shard
+		database, retentionPolicy, sgi := s.MetaStore.ShardOwner(req.ShardID())
```

Comment:
I think `ShardOwner`'s implementation should be changed such it does not return a ShardGroup that has been "deleted". If we don't do this, it's too easy to pass out shard groups that have been deleted, and the user doesn't check.

It seems to me that Shard Groups that have been deleted should be invisible to the system, except within the MetaStore. Otherwise we'll have bugs.